### DISCUSSION TITLE
Adding a "last edited" timestamp?

### DISCUSSION THREAD

**Discussion by alexgleason** [contentlayerdev/contentlayer#345] [Q&A] [answered]
Hi, I'm really enjoying using contentlayer! I've built a blog with Next.js and I'm planning to build documentation with it too.

Like some documentation websites, I'd like to add a "last edited" timestamp as well as a link to the page source on GitLab.

My current idea is to add it to computedFields under Post, and try to do git stuff in there to get the timestamp. Is this a sensible approach, or any other ideas?

**Comment by spcbfr** [comment]
I am also interested in this

**Comment by nkilm** [comment]
That would be nice! I'll be happy to help.

**Comment by alexgleason** [comment, marked as answer]
Hey guys, I figured it out.

See: [contentlayer.config.js](https://gitlab.com/soapbox-pub/soapbox.pub/-/blob/12d78beec6254657f622dac748ff63ee22b409de/contentlayer.config.js#L73-L80)

```js
    lastModified: {
      type: 'string',
      resolve: (post) => getFileInfo(`_blog/${post._raw.sourceFilePath}`).lastModified,
    },
    sourceUrl: {
      type: 'string',
      resolve: (post) => getFileInfo(`_blog/${post._raw.sourceFilePath}`).sourceUrl,
    },
```
[files.ts](https://gitlab.com/soapbox-pub/soapbox.pub/-/blob/12d78beec6254657f622dac748ff63ee22b409de/lib/files.ts):

```ts
import { getLastModified } from './git';

interface FileInfo {
  lastModified: string
  sourceUrl: string
}

const getFileInfo = (filename: string): FileInfo => {
  return {
    lastModified: getLastModified(filename),
    sourceUrl: `https://gitlab.com/soapbox-pub/soapbox.pub/-/blob/main/${filename}`,
  };
};

export {
  getFileInfo,
};
```

[git.ts](https://gitlab.com/soapbox-pub/soapbox.pub/-/blob/12d78beec6254657f622dac748ff63ee22b409de/lib/git.ts):

```ts
import { execFileSync } from 'child_process';

const getLastModified = (filename: string): string => {
  const raw = execFileSync('git', ['log', '-1', '--pretty=%cI', filename]);
  return String(raw).trim();
};

export {
  getLastModified,
};
```

Finally, [blog/[slug].tsx](https://gitlab.com/soapbox-pub/soapbox.pub/-/blob/12d78beec6254657f622dac748ff63ee22b409de/pages/blog/%5Bslug%5D.tsx#L46-L52):

```tsx
                {post.lastModified && (
                  <p className='text-gray-500 mt-12'>
                    Last edited <a className='text-azure' href={post.sourceUrl} target='_blank'>
                      <DateFormatter dateString={post.lastModified} />
                    </a>.
                  </p>
                )}
```
